meta.go

```diff
@@ -241,16 +242,90 @@ func parseStringList(list interface{}) []string {
 	return result
 }
 
-var termNameRE = regexp.MustCompile("^[a-z]+([a-z0-9-]+)(/[0-9]+)?$")
+var validTermName = regexp.MustCompile(`^[a-zA-Z][\w-]+$`)
 
-func checkTerm(s string) error {
-	match := termNameRE.FindStringSubmatch(s)
-	if match == nil {
-		return fmt.Errorf("invalid term name %q: must match %s", s, termNameRE.String())
+// Term represents a single term URI. The term can either be owned
+// or "public" (meaning there is no owner).
+type Term struct {
+	Tenant   string
+	Owner    string
+	Name     string
+	Revision int
+}
+
+// Validate returns an error if the Term contains invalid data.
+func (t *Term) Validate() error {
+	if t.Tenant != "" && t.Tenant != "cs" {
+		if !validTermName.MatchString(t.Tenant) {
+			return fmt.Errorf("wrong term tenant format %q", t.Tenant)
+		}
+	}
+	if t.Owner != "" && !names.IsValidUser(t.Owner) {
+		return fmt.Errorf("wrong owner format %q", t.Owner)
+	}
+	if !validTermName.MatchString(t.Name) {
+		return fmt.Errorf("wrong term name format %q", t.Name)
+	}
+	if t.Revision < 0 {
+		return fmt.Errorf("negative term revision")
 	}
 	return nil
 }
 
+// ParseTerm takes a term URI as a string and parses it into a Term.
+// A complete term is in the form:
+// tenant:owner/name/revision
+// This function accpets partial forms of this.
+// Typically in one of the following forms:
+// name
+// owner/name
+// owner/name/27 # Revision 27
+// name/283 # Revision 283
+// cs:owner/name # Tenant cs
+func ParseTerm(s string) (*Term, error) {
+	tenant := ""
+	termid := s
+	if t := strings.Split(s, ":"); len(t) == 2 {
```

Comment:
If there's an extra colon in there, we won't get a great error.
I'd suggest either doing SplitN(n, ":", 2) or returning an error if len(t) > 2.